component_library/scripts/build_scenario.py

```diff
@@ -87,6 +87,102 @@ def water_systems_postprocessing(self, survey):
         # This is a tradeoff between efficiency of survey processing and being able to treat special cases as we would
         # like to and make conditional choices (like add this component only to the drinking water bus and only if quesiton XYZ was answered with ...)
 
+    def waste_water_systems_postprocessing(self, survey):
+        """Go through the survey and implement specific logic regarding the water questions"""
+        # water_distinction_question_id = "7"
+        # waste_systems = survey["criteria_7"]
+        # if "septic system" in waste_systems:
+        # Add this to the list pass
+        # Need to be the same name as in WIP components in the csv
+        # self.components.update({"septic_system": {"name":"grey_water_se"}})
+        # self.components.update({"septic_system": {"name":"black_water_se"}})
+        # exemple if you need to change an attribute of a resource/component
+        # self.components["septic_system"].update({"capacity": 100})
+
+        #### Vivek's attempt at hardcode logic
+
+        # Survey responses assumed or taken from survey
+        wastewater_systems = survey["criteria_7"]
+        population = 1000 #  # survey needs to ask population, makes most of the logic implementation easier
+        toilet_types = survey["criteria_7.3"]
+
+        # bus addition
+        if "flush toilet" in toilet_types:
+            self.add_single_bus(name="bus-black-water", carrier="black-water")
+            self.add_single_bus(name="bus-grey-water", carrier="grey-water")
+        else:
+            self.add_single_bus(name="bus-grey-water", carrier="grey-water")
+
+
+
+
+
+        # black water treatment
+
+        if "flush toilet" in toilet_types:
+            if "septic system" in wastewater_systems:
+                self.components.update({("septic_system", "black_water_septic"): {"water_in_bus": "", "water_out_bus": ""}})
+            elif "constructed wetland" in wastewater_systems:
+                self.components.update({("constructed_wetland", "black_water_cw"): {"water_in_bus": "", "water_out_bus": ""}})
+
+
+        # grey water treatment
+
+        if "septic system" in wastewater_systems:
+            self.components.update({("septic_system", "grey_water_septic"): {"water_in_bus": "", "water_out_bus": ""}})
+
+        elif "constructed wetland" in wastewater_systems:
+            if "constructed_wetland" in self.components:
+                self.components["constructed_wetland"].update({"constructed_wetland": {"name": "grey_water_cw"}})
+            else:
+                self.components.update({"constructed_wetland": {"name": "grey_water_cw"}})
```

Comment:
```suggestion
            self.components.update({("constructed_wetland", "grey_water_cw"): {"water_in_bus": "", "water_out_bus": ""}})
```

Reply:
@Vivek-H-Rana - you can fix the calls below according to this suggestion (you turn 4 lines into one)